peekingduck/pipeline/nodes/model/jde_mot/jde_model.py

```diff
@@ -0,0 +1,182 @@
+            config["root"], config["weights"], config["weights_parent_dir"]
+        )
+        if not checker.has_weights(weights_dir, model_dir):
+            self.logger.info("No weights detected. Proceeding to download...")
+            downloader.download_weights(weights_dir, config["weights"]["blob_file"])
+            self.logger.info(f"Weights downloaded to {weights_dir}.")
+
+        self.config["cfg"] = str(model_dir / config["weights"]["config_file"])
+        self.config["weights_path"] = str(
+            model_dir / config["weights"]["model_file"]["jde"]
+        )
+        cfg_dict = parse_model_cfg(self.config["cfg"])
+        self.config["img_size"] = [
+            int(cfg_dict[0]["width"]),
+            int(cfg_dict[0]["height"]),
+        ]
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.tracker = JDETracker(self.config, self.device)
+
+    def predict(self, frame: np.ndarray) -> Tuple[np.ndarray, np.ndarray, List[str]]:
+        """Predicts person bboxes, object_id and labels.
+
+        Args:
+            frame (np.ndarray): image in numpy array
+
+        Returns:
+            bboxes (np.ndarray): numpy array of detected bboxes
+            obj_id (np.ndarray): numpy array of object tracking id
+            label (list): list of label of "person" for bboxes
+        """
+        img, img0 = self._reshape_image(frame)
+
+        # Run detection and tracking
+        blob = torch.from_numpy(img).unsqueeze(0).to(self.device)
+        online_targets = self.tracker.update(blob, img0)
+        online_tlwhs = []
+        online_ids = []
+        for t in online_targets:
+            tlwh = t.tlwh
+            tid = t.track_id
+            vertical = tlwh[2] / tlwh[3] > 1.6
+            if tlwh[2] * tlwh[3] > self.config["min_box_area"] and not vertical:
+                online_tlwhs.append(self._normalize_bboxes(tlwh))
+                online_ids.append(str(tid))
+
+        output_len = len(online_tlwhs)
+        labels = ["person"] * output_len
+        return online_tlwhs, online_ids, labels
+
+    def _reshape_image(self, frame: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """Reshape imput frame.
+
+        Args:
+            frame (np.ndarray): Frame to reshape
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray]: Normalized, resized frames.
+        """
+        width = self.config["img_size"][0]
+        height = self.config["img_size"][1]
+
+        vh, vw, _ = frame.shape
+        self.w, self.h = self._get_size(vw, vh, width, height)
+        img0 = cv2.resize(frame, (self.w, self.h))
+        # Padded resize
+        img, _, _, _ = letterbox(img0, height=height, width=width)
+
+        # Normalize RGB
+        img = img[:, :, ::-1].transpose(2, 0, 1)
+        img = np.ascontiguousarray(img, dtype=np.float32)
+        img /= 255.0
+        return img, img0
+
+    def _get_size(self, vw: int, vh: int, dw: int, dh: int) -> Tuple[int, int]:
```

Comment:
more meaningful argument names if possible

Reply:
Updated to more meaningful argument names